huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_app_rule_test.go

```diff
@@ -0,0 +1,213 @@
+						"C:\\Program Files\\Microsoft Office\\root\\Office16\\WINWORD.EXE"),
+					resource.TestCheckResourceAttr(withPathRule, "description", "Created by terraform script for path rule"),
+					resource.TestCheckResourceAttr(withPathRule, "rule.0.product_rule.#", "0"),
+				),
+			},
+			{
+				Config: testAccResourceAppRule_basic_step2(name),
+				Check: resource.ComposeTestCheckFunc(
+					rcWithProduct.CheckResourceExists(),
+					resource.TestCheckResourceAttr(withProductRule, "name", name),
+					resource.TestCheckResourceAttr(withProductRule, "rule.0.scope", "PRODUCT"),
+					resource.TestCheckResourceAttr(withProductRule, "rule.0.product_rule.0.identify_condition", "product"),
+					resource.TestCheckResourceAttr(withProductRule, "rule.0.product_rule.0.publisher", "Adobe Inc."),
+					resource.TestCheckResourceAttr(withProductRule, "rule.0.product_rule.0.product_name", "Adobe Photoshop"),
+					resource.TestCheckResourceAttr(withProductRule, "rule.0.product_rule.0.process_name", "Photoshop.exe"),
+					resource.TestCheckResourceAttr(withProductRule, "rule.0.product_rule.0.support_os", "Windows,Mac"),
+					resource.TestCheckResourceAttr(withProductRule, "rule.0.product_rule.0.version", "2.0"),
+					resource.TestCheckResourceAttr(withProductRule, "rule.0.product_rule.0.product_version", "2022"),
+					resource.TestCheckResourceAttr(withProductRule, "description", "Updated by terraform script"),
+					resource.TestCheckResourceAttr(withProductRule, "rule.0.path_rule.#", "0"),
+					rcWithPath.CheckResourceExists(),
+					resource.TestCheckResourceAttr(withPathRule, "name", name+"_path"),
+					resource.TestCheckResourceAttr(withPathRule, "rule.0.scope", "PATH"),
+					resource.TestCheckResourceAttr(withPathRule, "rule.0.path_rule.0.path",
+						"C:\\Program Files\\Adobe\\Adobe Photoshop 2022\\Photoshop.exe"),
+					resource.TestCheckResourceAttr(withPathRule, "description", "Updated by terraform script for path rule"),
+					resource.TestCheckResourceAttr(withPathRule, "rule.0.product_rule.#", "0"),
+				),
+			},
+			{
+				ResourceName:      withProductRule,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				ResourceName:      withPathRule,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config:      testAccResourceAppRule_validation_missing_identify_condition(name),
+				ExpectError: regexp.MustCompile(`The argument "identify_condition" is required, but no definition was found.`),
+			},
+			{
+				Config:      testAccResourceAppRule_validation_all_star_fields(name),
+				ExpectError: regexp.MustCompile(`publisher, product_name, and process_name cannot be all '\*' or empty in product_rule`),
+			},
+		},
+	})
+}
+
+func testAccResourceAppRule_basic_step1(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_workspace_app_rule" "with_product_rule" {
+  name = "%[1]s"
+  rule {
+    scope = "PRODUCT"
+    product_rule {
```

Comment:
```suggestion
    scope = "PRODUCT"

    product_rule {
```
validate this script format using hcbp-lint tool in your local main.tf file

Reply:
done